pkg/plugin/trivy/plugin.go

```diff
@@ -478,6 +495,21 @@ func (s *scanner) getPodSpecForClientServerMode(spec corev1.PodSpec, credentials
 			})
 		}
 
+		ref, err := name.ParseReference(container.Image)
+		if err != nil {
+			return corev1.PodSpec{}, nil, err
+		}
+		insecureRegistries := s.config.GetTrivyInsecureRegistries()
+		for _, insecureRegistry := range insecureRegistries {
+			if ref.Context().RegistryStr() == insecureRegistry {
+				env = append(env, corev1.EnvVar{
```

Comment:
We should have a unit test scenario which asserts that the TRIVY_INSECURE env is appended.

Reply:
added unit test